### PR TITLE
Add wait_for_cores_to_reach_state function.

### PR DESCRIPTION
**Pull request by mossblaser**
Also flake8-ify.

Also merges into #98 which in turn should be mergeable if this is OK!

### CODE REVIEW

**Comment by mossblaser** [comment]
Should probably add some tutorial docs... Getting on that. Don't merge just yet!

File: rig/machine_control/machine_controller.py [hunk 1]
```diff
@@ -995,6 +996,65 @@ def count_cores_in_state(self, state, app_id=Required):
...
+        """
+        if timeout is not None:
+            timeout_time = time.time() + timeout
+
+        while True:
+            cur_count = self.count_cores_in_state(state, app_id)
+            if cur_count >= count:
+                break
+
+            # Stop if timeout ellapsed
```
Comment (mundya) [inline rig/machine_control/machine_controller.py]: sp. elapsed

**Comment by mundya** [comment]
I'd definitely like something that included a: "is anything RTEd" check... but I guess that's trivial to add later.

**Comment by mossblaser** [comment]
Yeah... this is true, though as you say, I think I'd rather add this later since I'm not sure exactly how it should look.

Assuming these docs don't break the tests, everything should be good to go, I think!

**Pull request merged**